pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -259,7 +267,16 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 	registerHandlers(c.filteredNodeInformer, c.queue, "Nodes", c.onNodeEvent)
 
 	podInformer := sharedInformers.Core().V1().Pods().Informer()
-	c.pods = newPodCache(podInformer, c)
+	c.pods = newPodCache(podInformer, c, func(key string) {
+		item, exists, err := c.endpoints.getInformer().GetStore().GetByKey(key)
+		if err != nil || !exists {
```

Comment:
nit: In the latest we have logged error if it happens. Might be useful for debugging